Triviagen 7.1 Add rest of screens

Kazoroo opened the pull request:
I added 4 new screens: round setup, game screen with questions, results and screen with correct and checked answers. I created a package for every screen, where now is only the screen, but in the future there can be added viewModel or state class. For tests I use data from DummyTriviaRepositoryImpl. It have to be deleted and replaced with real viewModel and state when available. 

RoundSetupScreen:
![obraz](https://github.com/sidcgithub/ai-trivia-app-android/assets/91783342/5a305745-ff67-40ae-93f1-112cacb05888)

TriviaGameScreen:
![obraz](https://github.com/sidcgithub/ai-trivia-app-android/assets/91783342/afd07042-5457-4b59-8cb9-fa979f9d7c25)

ResultsScreen:
- when the score is below 51, the color of the circle is red.
![obraz](https://github.com/sidcgithub/ai-trivia-app-android/assets/91783342/d9c45223-002f-415d-b06b-f89951aa0863)

AnswersScreen:
![obraz](https://github.com/sidcgithub/ai-trivia-app-android/assets/91783342/b7ce51ad-2d47-4cc3-acb9-11c30beca9d9)

sidcgithub committed:
TRIVIAGEN-5: Defining the package structure for core and feature


Took 8 minutes

Took 49 minutes

sidcgithub committed:
TRIVIAGEN-5: Removed auth

Kazoroo committed:
Add interface in constructor instead of implementation.

Now it is repleceable in tests.

Rebase TRIVIAGEN-7 onto develop

Took 13 minutes

sidcgithub committed:
TRIVIAGEN-6: Added import and injection


Took 4 minutes

sidcgithub committed:
TRIVIAGEN-6: Tentative network wiring

Took 2 minutes

Commit:
Add new launcher icon

Commit:
Add app primary color, change typography, set app background color to brand color.

Took 2 hours 34 minutes

Commit:
Add main menu screen. Create reusable Scaffold function.

Took 10 minutes

Commit:
Fixes and improvments

Remove dynamic color, delete unused png, delete redundant Scaffold from MainActivity.kt, add dimens.xml and use those values in MainMenuScreen.

Took 24 minutes

Commit:
Optimize imports

Took 41 seconds

Commit:
Add screen for round setup. To TriviaGenScaffold add optional parameter for back arrow.

Also add required strings and dimensions to xmls

Took 1 hour 28 minutes

Commit:
Add screen for round setup. To TriviaGenScaffold add optional parameter for back arrow.

Also add required strings and dimensions to xmls

Took 1 hour 28 minutes

Commit:
Create TriviaGameScreen, optimize imports for RoundSetupScreen, change theme primary color, change parameter answer to integer in TriviaQuestion data class.

Took 59 minutes

Commit:
Create TriviaGameScreen, optimize imports for RoundSetupScreen, change theme primary color, change parameter answer to integer in TriviaQuestion data class.

Took 59 minutes

Commit:
Add ResultsScreen

Took 48 minutes

Commit:
Add ResultsScreen

Took 48 minutes

Commit:
Add screen for correct answers, also add file with card containing question and answers.

Took 1 hour 44 minutes

Commit:
Add screen for correct answers, also add file with card containing question and answers.

Took 1 hour 44 minutes

Commit:
Update dependencies version

Took 6 minutes

Commit:
Update dependencies version

Took 6 minutes

sidcgithub commented on app/src/main/java/com/triviagenai/triviagen/trivia/presentation/results/ResultsScreen.kt:
```diff
@@ -0,0 +1,108 @@
+            Box(
+                contentAlignment = Alignment.Center
+            ) {
+                CircularProgressIndicator(
```
This can be extracted to be it's own composable. Maybe something like `ScoreCircularBar()`

sidcgithub commented on app/src/main/java/com/triviagenai/triviagen/trivia/presentation/triviagame/TriviaGameScreen.kt:
```diff
@@ -0,0 +1,66 @@
+                        .padding(dimensionResource(id = R.dimen.padding_small))
+                        .width(dimensionResource(id = R.dimen.element_xlarge))
+                ) {
+                    Text(
```
Is the Option text vertically centered?

Kazoroo replied:
It's complicated. The small letters like ,,a" or ,,n" are in the middle, so from top edge of the letter is the same distance to the button border as from the bottom of letter. But in bigger letters like ,,b" or ,,V" it is different. They are higher, so if they want to be in the middle, as it is in the lower letters case, they would be a little below lower letters. And in my opinion it is some kind of viusal effect, that bigger letters generete and that is why it seems to be uncentered.

Green lines are the same height.
![obraz](https://github.com/sidcgithub/ai-trivia-app-android/assets/91783342/00cb7f2f-f5b6-4212-8dc1-6bb6cfe06ec2)

sidcgithub replied:
Hey makes sense.

sidcgithub commented on app/src/main/java/com/triviagenai/triviagen/trivia/presentation/roundsetup/RoundSetupScreen.kt:
```diff
@@ -0,0 +1,97 @@
+                    .width(dimensionResource(id = R.dimen.element_xlarge))
+                    .height(dimensionResource(id = R.dimen.element_height))
+            ) {
+                Text(
```
Is this text vertically centered?

sidcgithub replied:
Please ignore the centering of the text.

sidcgithub reviewed (commented):
Hey looks great! Just some minor comments. Not sure, but I noticed that text in the buttons does not appear to be vertically centered.

sidcgithub commented on app/src/main/java/com/triviagenai/triviagen/trivia/presentation/triviagame/TriviaGameScreen.kt:
```diff
@@ -0,0 +1,66 @@
+import com.triviagenai.triviagen.trivia.domain.model.TriviaQuestion
+
+@Composable
+fun TriviaGameScreen() {
```
Is the option to `Continue` missing?

Kazoroo replied:
Wouldn't it look like the user clicks an option and immediately sees the result?

sidcgithub replied:
The `TriviaGameScreen` will show up for as many questions that are there. The `ResultsScreen` would show up after all the questions are over. Also, clicking the right answer would put a light green shadow behind the option and clicking the wrong option would put a light red shadow behind that option.

Kazoroo replied:
Yeah but do we need this continue button? It can works like this: user click an option -> shaddow appears, change state -> the shadow stay for 2 seconds or more -> screen resets and user have new question

Can't we skip the special confirm button?

sidcgithub replied:
Now that I think of it, we'll be using the Speech engine. I guess for now we can remove the continue button and it can change automatically after 2 seconds. Later on, when the Speech engine announces "You've got it right!" or wrong we can automatically change after the TTS has completed. I'm okay with removing the button 👍

sidcgithub reviewed (approved):
LGTM

Commit:
Code readability improvements

Extract score progress bar to separate file, create a ComposableData that stores data for buttons or textfields, make a for loop for textfields in MainMenuScreen

Took 40 minutes

Commit:
Merge remote-tracking branch 'origin/test' into develop

Kazoroo committed:
Merge branch 'TRIVIAGEN-7-create-presentation-layer' into test
